File: trunk/resources/scop_1.5.1/examples/method_client.cpp
// method_client.cpp - DMI - 7-9-02

#include <scop.h>
#include <scopxml.h>

double cent_to_faren(int sock, double c);
double faren_to_cent(int sock, double f);
int count_uses(int sock);

int main()
{
	int sock;

	sock = scop_open("localhost", "method_client");
	
	printf("%g deg C = %g deg F.\n", 0.0, cent_to_faren(sock, 0.0));
	printf("%g deg C = %g deg F.\n", 20.0, cent_to_faren(sock, 20.0));
	printf("%g deg F = %g deg C.\n", 60.0, faren_to_cent(sock, 60.0));
	printf("The server has been accessed %d times.\n", count_uses(sock));
		
	close(sock);
	return 0;
}

double cent_to_faren(int sock, double c)
{
	vertex *v, *w;
	double f;
	
	v = pack(c);
	w = scop_rpc(sock, "method_server", v, "ctof");
	delete v;
	f = w->extract_double();
	delete w;
	return f;
}

double faren_to_cent(int sock, double f)
{
	vertex *v, *w;
	double c;
	
	v = pack(f);
	w = scop_rpc(sock, "method_server", v, "ftoc");
	delete v;
	c = w->extract_double();
	delete w;
	return c;
}

int count_uses(int sock)
{
	vertex *v, *w;
	int n;
	
	v = pack(0); // Dummy argument
	w = scop_rpc(sock, "method_server", v, "stats");
	n = w->extract_int();
	delete v;
	delete w;
	return n;
}

File: trunk/resources/scop_1.5.1/java/examples/MethodClient.java
// MethodClient.java - DMI - 25-12-02

import java.io.*;

public class MethodClient
{
	public static void main(String[] args)
	{
		SCOPXML scop;

		scop = new SCOPXML("localhost", "MethodClient");

		System.out.println(0.0 + " deg C = " + cent_to_faren(scop, 0.0) +
				" deg F.");
		System.out.println(20.0 + " deg C = " + cent_to_faren(scop, 20.0) +
				" deg F.");
		System.out.println(60.0 + " deg F = " + faren_to_cent(scop, 60.0) +
				" deg C.");
		System.out.println("The server has been accessed " +
				count_uses(scop) + " times.");

		scop.close();
	}

	private static double cent_to_faren(SCOPXML scop, double c)
	{
		Vertex v, w;
		double f;

		v = Vertex.pack(c);
		w = scop.rpc("MethodServer", v, "ctof");
		f = w.extract_double();
		return f;
	}

	private static double faren_to_cent(SCOPXML scop, double f)
	{
		Vertex v, w;
		double c;

		v = Vertex.pack(f);
		w = scop.rpc("MethodServer", v, "ftoc");
		c = w.extract_double();
		return c;
	}

	private static int count_uses(SCOPXML scop)
	{
		Vertex v, w;
		int n;

		v = Vertex.pack(0); // Dummy argument
		w = scop.rpc("MethodServer", v, "stats");
		n = w.extract_int();
		return n;
	}
}
